client/app/services/usage-graphs/usage-graphs.service.js

```diff
@@ -0,0 +1,79 @@
+/** @ngInject */
+export function UsageGraphsFactory() {
+  var service = {
+    cpuChartConfig: getCPUChartConfig,
+    memoryChartConfig: getMemoryChartConfig,
+    storageChartConfig: getStorageChartConfig,
+    convertBytestoGb: convertBytestoGb,
+  };
+
+  function getCPUChartConfig(used, total) {
+    let usedValue = 0;
+    let totalValue = 0;
+
+    if (angular.isDefined(used)) {
+      usedValue = used;
+      totalValue = total;
+    }
+
+    return {
+      config: {
+        units: __('MHz'),
+      },
+      data: {
+        'used': usedValue,
+        'total': totalValue,
+      },
+      label: __('used'),
+    };
+  }
+
+  function getMemoryChartConfig(used, total) {
+    let usedValue = 0;
+    let totalValue = 0;
+
+    if (angular.isDefined(used)) {
+      usedValue = used;
+      totalValue = total;
+    }
+
+    return {
+      config: {
+        units: __('GB'),
+        chartId: 'memoryChart',
+      },
+      data: {
+        'used': usedValue,
+        'total': totalValue,
+      },
+      label: __('used'),
+    };
+  }
+
+  function getStorageChartConfig(used, total) {
```

Comment:
Do we need a check here for a 0 total value so as to avoid a negative number? (same in the other methods too)